samtranslator/plugins/__init__.py

```diff
@@ -56,7 +56,7 @@ def on_before_transform_resource(self, logical_id, resource_type, resource_prope
         # NoOp implementation
         pass
 
-    def on_before_transform_template(self, template):
+    def on_before_transform_template(self, template_dict):
```

Comment:
Why?

Reply:
This is changed due to pyling rule W0237. 
```
Used when a method parameter has a different name than in the implemented interface or in an overridden method. 
```
https://pylint.pycqa.org/en/latest/user_guide/messages/warning/arguments-renamed.html